src/writers/sink.creator.cpp

```diff
@@ -81,7 +81,7 @@ zarr::SinkCreator::make_dirs_(std::queue<std::string>& dir_paths)
         return true;
     }
 
-    std::atomic<bool> all_successful = true;
+    std::atomic<char> all_successful = 1;
```

Comment:
Hmm, why is it necessary to change the underlying type to `char`?  I thought that `std::atomic<bool>` implements  `fetch_and`.

Reply:
Not according to my compiler.